packages/mui-material-next/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -3,30 +3,42 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import { unstable_composeClasses as composeClasses } from '@mui/base/composeClasses';
-import { alpha } from '@mui/system';
+import { CSSInterpolation, alpha } from '@mui/system';
 import { getValidReactChildren, unstable_capitalize as capitalize } from '@mui/utils';
 import styled from '../styles/styled';
 import useThemeProps from '../styles/useThemeProps';
 import buttonGroupClasses, { getButtonGroupUtilityClass } from './buttonGroupClasses';
 import ButtonGroupContext from './ButtonGroupContext';
 import ButtonGroupButtonContext from './ButtonGroupButtonContext';
+import { ButtonGroupOwnerState, ButtonGroupProps, ButtonGroupTypeMap } from './ButtonGroup.types';
 
-const overridesResolver = (props, styles) => {
-  const { ownerState } = props;
+const overridesResolver = (
```

Comment:
You can just inline this `overridesResolver` like this: https://github.com/mui/material-ui/blob/master/packages/mui-material-next/src/Chip/Chip.tsx#L45-L64

That way you shouldn't need to type the arguments manually ~